feat(sign-in-with-google): create the Google SignIn feature #11

ALVINdimpos opened the pull request:
What does this PR do?

       This PR adds the ability for users to log in using their Google account.

    Description of Task to be completed

        Implement Google OAuth2 authentication in the backend
        Create a new endpoint for Google login and handle the login process
        Store the user's Google ID and access token securely in the database
        Ensure the user is redirected to the correct page after logging in
        Implement the ability for users to log out of their Google account and log in again using the same credentials
        Document the implementation following best practices for security and maintainability

    How should this be manually tested?

         npm install
         npm run dev
        then follow this route http://localhost:3000/api/auth/google/callback

    Any background context you want to provide?

        This feature was requested by users who prefer to log in using their Google accounts rather than creating a new                                         email/password account.
        What are the relevant pivotal tracker stories?

        CardId #11: Add Google login feature
        CardLink:https://trello.com/c/FQfu8x6Y/11-wip-sign-in-with-social-accounts

    Screenshots (if appropriate)

    N/A (or add relevant screenshots if appropriate)

ALVINdimpos committed:
feat(sign-in-with-google): create the Google SignIn feature

- Create a new route for handling Google authentication and redirecting the user to Google's OAuth2 consent screen.
- Handle the callback from Google and added documentation

[Finishes #11]

ALVINdimpos committed:
feat(sign-in-with-google): create the Google SignIn feature

- Create a new route for handling Google authentication and redirecting the user to Google's OAuth2 consent screen.
- Handle the callback from Google and added documentation

[Finishes #11]

ALVINdimpos committed:
feat(sign-in-with-google): create the Google SignIn feature

- Create a new route for handling Google authentication and redirecting the user to Google's OAuth2 consent screen.
- Handle the callback from Google and added documentation

[Finishes #11]

ALVINdimpos commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[WIP] Sign In with Social Accounts](https://trello.com/c/FQfu8x6Y/11-wip-sign-in-with-social-accounts)

Muhet reviewed (commented):
keep it up man, good work

chanceDira commented on .github/workflows/node.js.yml:
```diff
@@ -24,17 +24,4 @@ jobs:
-          retries: 20
-          wait: 16000
-          sleep: 30000
+  
```
Why deploy job is removed?

ALVINdimpos replied:
i have remove it becouse it require everyone to deploy their own repo in order to pass check and also to set their own RENDER_TOKEN

chanceDira commented on src/controller/authController.js:
```diff
@@ -0,0 +1,97 @@
+export const GetUsers = async (req, res) => {
+  try {
+    const users = await db.User.findAll();
+    return res.status(200).json({ users });
```
Let's use Jsend format to structure our responses.

ALVINdimpos replied:
done!

chanceDira commented on src/controller/authController.js:
```diff
@@ -0,0 +1,97 @@
+export const GetUsers = async (req, res) => {
+  try {
+    const users = await db.User.findAll();
+    return res.status(200).json({ users });
```
Let's use Jsend format to structure our responses.

ALVINdimpos replied:
done!

chanceDira reviewed (changes requested):
Nice work @ALVINdimpos, I just added some feedback for you.

Also, let's use this PR example for team consistency 
https://github.com/atlp-rwanda/engineering-playbook/wiki/3.-Pull-Request

bryan936M reviewed (approved):
Keep up the good work.

yobuir reviewed (approved):
Great job @ALVINdimpos

faid-terence reviewed (approved):
Good job @ALVINdimpos

NTElissa reviewed (approved):
LGTM

ALVINdimpos commented:
> LGTM

what do u mean sir?

ALVINdimpos committed:
feat(sign-in-with-google): create the Google SignIn feature

- Create a new route for handling Google authentication and redirecting the user to Google's OAuth2 consent screen.
- Handle the callback from Google and added documentation

[Finishes #11]

ALVINdimpos committed:
feat(sign-in-with-google): create the Google SignIn feature

- Create a new route for handling Google authentication and redirecting the user to Google's OAuth2 consent screen.
- Handle the callback from Google

[Finishes #11]

Muhet reviewed (approved):
well done broh

Commit:
feat(signup): buyer registration

- add route to register user in app
- create route, controller for endpoint
- create documentation for endpoint

[Delivers #3]

chanceDira reviewed (approved):
Good work @ALVINdimpos
